crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -207,8 +207,8 @@ fn infer_unpack_types<'db>(db: &'db dyn Db, unpack: Unpack<'db>) -> UnpackResult
     let result = infer_expression_types(db, value);
     let value_ty = result.expression_ty(value.node_ref(db).scoped_expression_id(db, scope));
 
-    let mut unpacker = Unpacker::new(db, file);
-    unpacker.unpack(unpack.target(db), value_ty, scope);
+    let mut unpacker = Unpacker::new(db, file, scope);
```

Comment:
Nit: It should be enough to only pass `scope` because you can get to the file by calling `scope.file(db)`